### PR TITLE
feat(flex-stacker): implement platform sensors and door switch gpios

### PR DESCRIPTION
**Pull request by ahiuchingau**
This PR enables the reading of the hopper door switch and platform sensors from the Flex Stacker.

My logic for placing the platform sensors in the motor task is that the sensors will be primarily used for motor movements. The door switch logic is owned by the system task instead.

### CODE REVIEW

File: stm32-modules/include/flex-stacker/flex-stacker/messages.hpp [hunk 1]
```diff
@@ -239,6 +239,27 @@ struct GetMotorStallGuardResponse {
     int sgt;
 };
 
+struct GetDoorClosedMessage {
+    uint32_t id;
+};
+
+struct GetDoorClosedResponse {
+    uint32_t responding_to_id;
+    ;
```
Comment (vegano1) [inline stm32-modules/include/flex-stacker/flex-stacker/messages.hpp]: typo

File: stm32-modules/include/flex-stacker/flex-stacker/messages.hpp [hunk 2]
```diff
@@ -239,6 +239,27 @@ struct GetMotorStallGuardResponse {
...
+};
+
+struct GetPlatformSensorsMessage {
+    uint32_t id;
+};
+
+struct GetPlatformSensorsResponse {
+    uint32_t responding_to_id;
+    ;
+    bool extend;   // Sensor located on the positive end of X axis
```
Comment (vegano1) [inline stm32-modules/include/flex-stacker/flex-stacker/messages.hpp]: remove ;
`extend` is a keyword

**Review by vegano1 — APPROVED** [review]
nice, this looks good.
left 2 small comments.

**Review by sfoster1 — APPROVED** [review]
agree with the semicolon and keyword things, looks good otherwise

**Pull request merged**